src/shared/components/query/QueryStore.ts

```diff
@@ -297,8 +230,28 @@ export class QueryStore
 	{
 		let ids:string[] = this._selectedStudyIds.keys();
 		const selectableStudies = this.selectableStudiesSet;
-		ids = ids.filter(id=>!!selectableStudies[id]);
-		return this.forDownloadTab ? ids.slice(-1) : ids;
+		let selectableIds = ids.filter(id=>!!selectableStudies[id]);

```

Comment:
this logic could use some commenting.  someone coming to it without background would struggle to understand what is going on with "selectable" studies and also the normalization.  (I do realize a lot of this is legacy code from another author but no time like the present to make it easier to understand)

Reply:
True, I should have added some comments here. Now I don't remember why I added this block. For now I'll revert it back.